### ISSUE TITLE
CloudFormation integration

### ISSUE THREAD

**Issue by tj** [apex/apex#516]
Not bailing on Terraform but CloudFormation's change sets seem to work pretty well, that was a major blocker before, but it could be great to add some sort of CFN support.

**Comment by DavidTPate** [comment]
Yeah, I get how CFN could lead to a better user experience as well. I think there will be some complications around there when dealing with Parent/Child templates and other pieces.

Probably need to put together a facade for how things like Role names are pulled in and other pieces so that it can be standardized.
